backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHIteratorApi.scala

```diff
@@ -97,16 +97,26 @@ class CHIteratorApi extends IteratorApi with Logging with LogLevelUtil {
    */
   override def genPartitions(
       wsCtx: WholeStageTransformContext,
-      splitInfos: Seq[Seq[SplitInfo]]): Seq[BaseGlutenPartition] = {
+      splitInfos: Seq[Seq[SplitInfo]],
+      scans: Seq[BasicScanExecTransformer]): Seq[BaseGlutenPartition] = {
+    // Only serialize plan once, save lots time when plan is complex.
+    val planByteArray = wsCtx.root.toProtobuf.toByteArray
     splitInfos.zipWithIndex.map {
-      case (splitInfos, index) =>
-        wsCtx.substraitContext.initSplitInfosIndex(0)
-        wsCtx.substraitContext.setSplitInfos(splitInfos)
-        val substraitPlan = wsCtx.root.toProtobuf
+      case (splits, index) =>
+        val splitInfosByteArray = splits.zipWithIndex.map {
+          case (split, i) =>
+            val filesNode = split.asInstanceOf[LocalFilesNode]
+            filesNode.setFileSchema(scans(i).getDataSchema)
```

Comment:
@Yohahaha It seems LocalFilesNode doesn't set fileSchema and fileReadProperties in velox after decouping LocalFiles from plan, could you please check and verify whether velox needs it?

Reply:
Velox use `base_schema` defined in ReadRel, instead of `schema` defined in LocalFiles.FileOrFiles, one read rel should only has one schema I think.

`fileReadProperties` is not needed in Velox for now.